pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -40,11 +39,17 @@ import (
 	"istio.io/istio/pkg/kube/controllers"
 	kubelabels "istio.io/istio/pkg/kube/labels"
 	"istio.io/istio/pkg/maps"
+	"istio.io/istio/pkg/slices"
 	"istio.io/istio/pkg/spiffe"
 	"istio.io/istio/pkg/util/sets"
 	"istio.io/istio/pkg/workloadapi"
 )
 
+const (
+	workloadEntryGroupKind = "/networking.istio.io/WorkloadEntry/"
+	serviceEntryGroupKind  = "/networking.istio.io/ServiceEntry/"
```

Comment:
please donot invent a new Kind, we should always obey the k8s schema

Reply:
yeah i think we can skip looking at the UID itself altogether, not relying on the format at all if we did https://github.com/istio/istio/pull/46060#discussion_r1268159766